test/__init__.py

```diff
@@ -231,35 +239,37 @@ def wrapper(*args, **kwargs):
 
 
 class _ListHandler(logging.Handler):
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__()
-        self.records = []
+        self.records: List[logging.LogRecord] = []
 
-    def emit(self, record):
+    def emit(self, record: logging.LogRecord) -> None:
         self.records.append(record)
 
 
 class LogRecorder:
-    def __init__(self, target=logging.root):
+    def __init__(self, target: logging.Logger = logging.root) -> None:
         super().__init__()
         self._target = target
         self._handler = _ListHandler()
 
     @property
-    def records(self):
+    def records(self) -> List[logging.LogRecord]:
         return self._handler.records
 
-    def install(self):
+    def install(self) -> None:
         self._target.addHandler(self._handler)
 
-    def uninstall(self):
+    def uninstall(self) -> None:
         self._target.removeHandler(self._handler)
 
-    def __enter__(self):
+    def __enter__(self) -> List[logging.LogRecord]:
         self.install()
         return self.records
 
-    def __exit__(self, exc_type, exc_value, traceback):
+    def __exit__(
+        self, exc_type: object, exc_value: object, traceback: object
```

Comment:
If you want to be pedantic, this can be something like `exc_type: Type[BaseException], exc_value: BaseException, traceback: types.TracebackType`. Hmm or actually, with `Optional` around all of them...